vaccine/vaccine_reg_ussd.py

```diff
@@ -0,0 +1,363 @@
+        )
+
+    async def state_dob_month(self):
+        return ChoiceState(
+            self,
+            question="DOB: Select your month",
+            choices=[
+                Choice("1", "Jan"),
+                Choice("2", "Feb"),
+                Choice("3", "Mar"),
+                Choice("4", "Apr"),
+                Choice("5", "May"),
+                Choice("6", "June"),
+                Choice("7", "July"),
+                Choice("8", "Aug"),
+                Choice("9", "Sep"),
+                Choice("10", "Oct"),
+                Choice("11", "Nov"),
+                Choice("12", "Dec"),
+            ],
+            next="state_dob_day",
+            error="REQUIRED: In which month were you born?",
+        )
+
+    async def state_dob_day(self):
+        # TODO: stop <40 year olds from continuing
+        async def validate_dob_day(value):
+            dob_year = int(self.user.answers["state_dob_year"])
+            dob_month = int(self.user.answers["state_dob_month"])
+            try:
+                assert value.isdigit()
+                date(dob_year, dob_month, int(value))
+            except (AssertionError, ValueError):
+                raise ErrorMessage(
+                    "ERROR: Please reply with DAY of your birthday Example: 20"
+                )
+
+        return FreeText(
+            self,
+            question="DOB: Which day of the month were you born on",
+            next="state_first_name",
+            check=validate_dob_day,
+        )
+
+    async def state_first_name(self):
+        return FreeText(
+            self, question="Please enter your FIRST name", next="state_surname"
+        )
+
+    async def state_surname(self):
+        return FreeText(
+            self, question="Please enter your SURNAME", next="state_confirm_profile"
+        )
+
+    async def state_confirm_profile(self):
+        first_name = self.user.answers["state_first_name"][:27]
+        surname = self.user.answers["state_surname"][:27]
+        id_type = self.ID_TYPES[self.user.answers["state_identification_type"]].value
+        id_number = self.user.answers["state_identification_number"][:20]
+        return MenuState(
+            self,
+            question="\n".join(
+                [
+                    "Confirm the following:",
+                    "",
+                    f"{first_name} {surname}",
+                    id_type,
+                    id_number,
+                ]
+            ),
+            choices=[
+                Choice("state_province", "Yes"),
+                Choice("state_identification_type", "No"),
+            ],
+            error="\n".join(
+                [
+                    "Is the information you shared correct?",
+                    "",
+                    f"{first_name} {surname}",
+                    id_type,
+                    id_number,
+                ]
+            ),
+        )
+
+    async def state_province(self):
+        return ChoiceState(
```

Comment:
Add `#TODO` for using the Province selector using the EVDS uuids for Province. Alternatively map them in the end

Reply:
Looking at the API, it seems like all they want is the lowest tier result (ie. suburb), and then they infer the parent fields from that. But I'll add a todo here, since we'll need the province ID to filter the suburbs.